Src/WitsmlExplorer.Api/Services/TrajectoryService.cs

```diff
@@ -42,24 +46,54 @@ public async Task<IEnumerable<Trajectory>> GetTrajectories(string wellUid, strin
                 }).OrderBy(trajectory => trajectory.Name);
         }
 
+        public async Task<Trajectory> GetTrajectory(string wellUid, string wellboreUid, string trajectoryUid)
+        {
+            var witsmlTrajectory = TrajectoryQueries.GetWitsmlTrajectoryById(wellUid, wellboreUid, trajectoryUid);
+            var result = await WitsmlClient.GetFromStoreAsync(witsmlTrajectory, new OptionsIn(ReturnElements.All));
+
+            if (result.Trajectories.Any())
+            {
+                return WitsmlToTrajectory(result.Trajectories.First());
+            }
+
+            return null;
+        }
+
         public async Task<List<TrajectoryStation>> GetTrajectoryStations(string wellUid, string wellboreUid, string trajectoryUid)
         {
             var trajectoryToQuery = TrajectoryQueries.GetWitsmlTrajectoryById(wellUid, wellboreUid, trajectoryUid);
             var result = await WitsmlClient.GetFromStoreAsync(trajectoryToQuery, new OptionsIn(ReturnElements.All));
             var witsmlTrajectory = result.Trajectories.FirstOrDefault();
             if (witsmlTrajectory == null) return null;
             return witsmlTrajectory.TrajectoryStations.Select(tStation => new TrajectoryStation
-                {
-                    Uid = tStation.Uid,
-                    DTimStn = StringHelpers.ToDateTime(tStation.DTimStn),
-                    TypeTrajStation = tStation.TypeTrajStation,
-                    Md = StringHelpers.ToDecimal(tStation.Md?.Value),
-                    Tvd = StringHelpers.ToDecimal(tStation.Tvd?.Value),
-                    Incl = StringHelpers.ToDecimal(tStation.Incl?.Value),
-                    Azi = StringHelpers.ToDecimal(tStation.Azi?.Value)
-                })
-                .OrderBy(tStation => tStation.Md)
+            {
+                Uid = tStation.Uid,
+                DTimStn = StringHelpers.ToDateTime(tStation.DTimStn),
+                TypeTrajStation = tStation.TypeTrajStation,
+                Md = (tStation.Md == null) ? null : new LengthMeasure { Uom = tStation.Md.Uom, Value = decimal.Parse(tStation.Md.Value) },
+                Tvd = (tStation.Tvd == null) ? null : new LengthMeasure { Uom = tStation.Tvd.Uom, Value = decimal.Parse(tStation.Tvd?.Value) },
+                Incl = (tStation.Incl == null) ? null : new LengthMeasure { Uom = tStation.Incl.Uom, Value = decimal.Parse(tStation.Incl?.Value) },
+                Azi = (tStation.Azi == null) ? null : new LengthMeasure { Uom = tStation.Azi.Uom, Value = decimal.Parse(tStation.Azi?.Value) }
+            })
+                .OrderBy(tStation => tStation.Md.Value)
                 .ToList();
         }
+        private static Trajectory WitsmlToTrajectory(WitsmlTrajectory trajectory)
+        {
+            return new Trajectory
```

Comment:
Use WitsmlToTrajectory in GetTrajectories as well to avoid code duplication

Reply:
fixed now